File: encoders/qls/src/test/java/nl/uva/sc/encoders/qls/ast/PageBuilder.java
package nl.uva.sc.encoders.qls.ast;

import static nl.uva.sc.encoders.ql.ast.TextLocationBuilder.aTextLocation;
import static nl.uva.sc.encoders.qls.ast.SectionBuilder.aSection;

import java.util.ArrayList;
import java.util.List;

import nl.uva.sc.encoders.ql.ast.TextLocation;

public class PageBuilder {

	private TextLocation textLocation;
	private String name;
	private List<Section> sections = new ArrayList<Section>();
	private List<DefaultStyle> pageDefaultStyles = new ArrayList<>();

	public static PageBuilder aPage() {
		PageBuilder pageBuilder = new PageBuilder();
		pageBuilder.textLocation = aTextLocation().build();
		pageBuilder.name = "SomePage";
		pageBuilder.sections.add(aSection().build());
		return pageBuilder;
	}

	public Page build() {
		return new Page(textLocation, name);
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Building/PageBuilder.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Building;
using UvA.SoftCon.Questionnaire.QLS.AST.Model;
using UvA.SoftCon.Questionnaire.QLS.Grammar;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Building
{
    internal class PageBuilder : QLSBaseVisitor<Page>
    {
        public override Page VisitPage(QLSParser.PageContext context)
        {
            var id = new Identifier(context.ID().GetText(), context.GetTextPosition());
            var sections = new List<Section>();
            var defaultStyles = new List<DefaultStyle>();

            foreach (var sectionContext in context.section())
            {
                sections.Add(sectionContext.Accept(new SectionBuilder()));
            }
            foreach (var defaultStyleContext in context.default_style())
            {
                defaultStyles.Add(defaultStyleContext.Accept(new DefaultStyleBuilder()));
            }

            return new Page(id, sections, defaultStyles, context.GetTextPosition());
        }
    }
}
